### PR TITLE
Add supplier account scenarios

### PR DESCRIPTION
**Pull request by TheDoubleK**
"AA" Trello card: https://trello.com/c/IfLi5jKl/59-port-supplier-account-functional-tests-from-legacy-to-main-branch
For this tech-debt card: https://trello.com/c/0bFSSGP8/2-rewriting-functional-tests

Replaces all the scenarios from the 'legacy' branch here (including two "WIP" scenarios that have never run but that we now need as we're finally turning on supplier editing of G-Cloud services): https://github.com/alphagov/digitalmarketplace-functional-tests/blob/jenkins/features/supplier/supplier_journey.feature

In doing this I've generalised a couple of step definitions:
 * "I have a live digital outcomes and specialists framework" is now "I have a live <framework-slug> framework".
* "I see the '(.*)' summary table filled with..." now looks for the table caption rather than table heading, to allow matching summary tables without a heading.

I've also added a Cloud Support service fixture to allow testing some G-Cloud stuff.  We can add the other G-Cloud lots if we ever need them, but a service on one G-Cloud lot will suffice for most things I think (certainly for everything here).

When this gets merged we should merge this at the same time:
 - [ ] https://github.com/alphagov/digitalmarketplace-functional-tests/pull/392

### CODE REVIEW

File: features/support/fixtures.rb [hunk 1]
```diff
@@ -238,4 +238,48 @@ def self.digital_specialists_brief_response
...
+          "Feature 1"
+      ],
+      serviceName: "Test cloud support service",
+      setupAndMigrationService: false,
+      sfiaRateDocumentURL: "https://assets.digitalmarketplace.service.gov.uk/g-cloud-9/documents/",
+      staffSecurityClearanceChecks: "staff_screening_not_bs7858_2012",
+      supportLevels: "None",
+      termsAndConditionsDocumentURL: "https://assets.digitalmarketplace.service.gov.uk/g-cloud-9/documents/",
+      training: false,
+      webChatSupport: "no",
```
Comment (katstevens) [inline features/support/fixtures.rb]: I am going to trust you that these are all the right cloud support fields, as I don't have a clue 👍

File: features/supplier/view_and_edit_g_cloud_services.feature [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+Scenario: Supplier can remove their G-Cloud service from the marketplace
+  When I click 'Remove service'
+  Then I see a destructive banner message containing 'Are you sure you want to remove your service?'
+
+  When I click 'Remove service'
+  Then I see a success banner message containing 'Test cloud support service has been removed.'
+
+  When I click 'Test cloud support service'
+  Then I see a temporary-message banner message containing 'This service was removed'
```
Comment (katstevens) [inline features/supplier/view_and_edit_g_cloud_services.feature]: Nice. Will this ensure that our setup step in the Background will only find one service?

File: features/step_definitions/emails.rb [hunk 3]
```diff
@@ -29,3 +29,9 @@ def self.get_email(message_type, email_address)
   @email_address = randomString + '@example.gov.uk'
   puts "Email address: #{@email_address}"
 end
+
+Given /^I have a random email address$/ do
+  randomString = SecureRandom.hex
+  @email_address = randomString + '@example.com'
+  puts "Email address: #{@email_address}"
+end
```
Comment (katstevens) [inline features/step_definitions/emails.rb]: An EOF char has sneaked in here...

File: features/support/fixtures.rb (hunk 1 above)
Comment (TheDoubleK) [reply features/support/fixtures.rb]: Yeah I got one from the API and redacted values where necessary.

File: features/supplier/view_and_edit_g_cloud_services.feature (hunk 2 above)
Comment (TheDoubleK) [reply features/supplier/view_and_edit_g_cloud_services.feature]: Yeah only one service is ever added for the supplier.

File: features/step_definitions/emails.rb (hunk 3 above)
Comment (TheDoubleK) [reply features/step_definitions/emails.rb]: Or rather, a missing line terminator.  Will fix.

**Pull request merged**